shared/data/web_accessible_resources/fb-sdk.js

```diff
@@ -0,0 +1,144 @@
+//facebook.net/sdk.js application/javascript
+(() => {
+    'use strict'
+    const originalFBURL = document.currentScript.src
+    let siteInit = function () {}
+    let fbIsEnabled = false
+    let initData = {}
+    const parseCalls = []
+    const popupName = Math.random().toString(36).replace(/[^a-z]+/g, '').substr(0, 12)
+
+    const fbLogin = {
+        callback: function () {},
+        params: undefined,
+        shouldRun: false
+    }
+
+    function messageAddon (detailObject) {
+        detailObject.entity = 'Facebook'
+        const event = new CustomEvent('ddgClickToLoad', {
+            detail: detailObject,
+            bubbles: false,
+            cancelable: false,
+            composed: false
+        })
+        dispatchEvent(event)
+    }
+
+    function enableFacebookSDK () {
+        if (!fbIsEnabled) {
+            window.FB = undefined
+
+            window.fbAsyncInit = function () {
+                if (initData) {
+                    window.FB.init(initData)
+                }
+                siteInit()
+                if (fbLogin.shouldRun) {
+                    window.FB.login(fbLogin.callback, fbLogin.params)
+                }
+            }
+
+            const fbScript = document.createElement('script')
+            fbScript.src = originalFBURL
+            fbScript.onload = function () {
+                for (const node of parseCalls) {
+                    window.FB.XFBML.parse.apply(window.FB.XFBML, node)
+                }
+            }
+            document.head.appendChild(fbScript)
+            fbIsEnabled = true
+        }
+    }
+
+    function runFacebookLogin () {
+        fbLogin.shouldRun = true
+        replaceWindowOpen()
+        loginPopup()
+        enableFacebookSDK()
+    }
+
+    function replaceWindowOpen () {
+        const oldOpen = window.open
+        window.open = function (url, name, windowParams) {
+            const u = new URL(url)
+            if (u.origin === 'https://www.facebook.com') {
+                name = popupName
+            }
+            return oldOpen.call(window, url, name, windowParams)
+        }
+    }
+
+    function loginPopup () {
+        const width = Math.min(window.screen.width, 450)
+        const height = Math.min(window.screen.height, 450)
+        const popupParams = `width=${width},height=${height},scrollbars=1,location=1`
+        window.open('about:blank', popupName, popupParams)
+    }
+
+    window.addEventListener('LoadFacebookSDK', enableFacebookSDK)
+    window.addEventListener('RunFacebookLogin', runFacebookLogin)
```

Comment:
They can emit these events and trigger the behavior, but it won't actually enable FB calls. That's because the flow requires a user click before it sends a message to the extension to actually allow FB traffic. In the worst case, a site falsely generating these events will break the surrogate function, as the surrogate will try to unsuccessfully load the real SDK.